src/custom-element.js

```diff
@@ -123,92 +123,103 @@ export function stubElements(win) {
  * @param {!AmpElement} element
  */
 export function applyLayout_(element) {
-  let widthAttr = element.getAttribute('width');
-  let heightAttr = element.getAttribute('height');
-  const sizesAttr = element.getAttribute('sizes');
   const layoutAttr = element.getAttribute('layout');
+  const widthAttr = element.getAttribute('width');
+  const heightAttr = element.getAttribute('height');
+  const sizesAttr = element.getAttribute('sizes');
+
+  // Input layout attributes.
+  const inputLayout = layoutAttr ? parseLayout(layoutAttr) : null;
+  assert(inputLayout || !layoutAttr, 'Unknown layout: %s', layoutAttr);
```

Comment:
Yes. Asserts are fine. In this case I preferred it because the expressions look a lot cleaner.

The `inputLayout !== undefined` is equivalent. I'll switch to it.